diff --git a/clspv/lib/BitcastUtils.cpp b/clspv/lib/BitcastUtils.cpp
index a4284dd8..77228209 100644
--- a/clspv/lib/BitcastUtils.cpp
+++ b/clspv/lib/BitcastUtils.cpp
@@ -384,8 +384,9 @@ void GroupVectorValuesInPair(IRBuilder<> &Builder,
 
   for (unsigned i = 0; i < NewValuesSize; i++) {
     unsigned idx = 2 * i;
+    SmallVector<unsigned, 4> Indices = {0, 1, 2, 3};
     Values[i] =
-        Builder.CreateShuffleVector(Values[idx], Values[idx + 1], {0, 1, 2, 3});
+        Builder.CreateShuffleVector(Values[idx], Values[idx + 1], Indices);
   }
   Values.resize(NewValuesSize);
 }
diff --git a/clspv/lib/Compiler.cpp b/clspv/lib/Compiler.cpp
index 89bcc675..dc8c3f47 100644
--- a/clspv/lib/Compiler.cpp
+++ b/clspv/lib/Compiler.cpp
@@ -272,7 +272,6 @@ int SetCompilerInstanceOptions(
   instance.getCodeGenOpts().SimplifyLibCalls = false;
   instance.getCodeGenOpts().EmitOpenCLArgMetadata = false;
   instance.getCodeGenOpts().DisableO0ImplyOptNone = true;
-  instance.getCodeGenOpts().OpaquePointers = clspv::Option::OpaquePointers();
   instance.getDiagnosticOpts().IgnoreWarnings = IgnoreWarnings;
 
   instance.getLangOpts().SinglePrecisionConstants =
@@ -320,8 +319,9 @@ int SetCompilerInstanceOptions(
   // We manually include the OpenCL headers below, so this vector is unused.
   std::vector<std::string> includes;
 
-  LangOptions::setLangDefaults(instance.getLangOpts(), clang::Language::OpenCL,
-                               triple, includes, standard);
+  instance.getInvocation().setLangDefaults(
+      instance.getLangOpts(), clang::InputKind(clang::Language::OpenCL), triple,
+      includes, standard);
 
   // Override the C99 inline semantics to accommodate for more OpenCL C
   // programs in the wild.
diff --git a/clspv/lib/PushConstant.cpp b/clspv/lib/PushConstant.cpp
index a1ba95a4..49b1344f 100644
--- a/clspv/lib/PushConstant.cpp
+++ b/clspv/lib/PushConstant.cpp
@@ -259,7 +259,10 @@ void RedeclareGlobalPushConstants(Module &M, StructType *mangled_struct_ty,
         auto new_gep = ConstantExpr::getGetElementPtr(
             push_constant_ty, new_GV, indices, gep_operator->isInBounds());
         user->replaceAllUsesWith(new_gep);
+      } else if (auto ptrtoint = dyn_cast<PtrToIntOperator>(user)) {
+        user->replaceAllUsesWith(ConstantExpr::getPtrToInt(new_GV, ptrtoint->getType()));
       } else {
+        user->print(errs());
         assert(false && "unexpected global use");
       }
     }
