diff --git a/clvk/src/config.def b/clvk/src/config.def
index 15e945f..20b66f5 100644
--- a/clvk/src/config.def
+++ b/clvk/src/config.def
@@ -14,8 +14,8 @@
 
 OPTION(std::string, cache_dir, "")
 OPTION(uint32_t, max_cmd_batch_size, 10000u)
-OPTION(uint32_t, max_first_cmd_batch_size, 10000u)
-OPTION(uint32_t, max_cmd_group_size, UINT32_MAX)
+OPTION(uint32_t, max_first_cmd_batch_size, 10u)
+OPTION(uint32_t, max_cmd_group_size, 1u)
 OPTION(uint32_t, max_first_cmd_group_size, UINT32_MAX)
 OPTION(bool, skip_spirv_capability_check, false)
 OPTION(bool, validation_layers, false)
