diff --git a/clvk/src/device.cpp b/clvk/src/device.cpp
index 6374490..38f9c1c 100644
--- a/clvk/src/device.cpp
+++ b/clvk/src/device.cpp
@@ -497,7 +497,6 @@ void cvk_device::init_compiler_options() {
     m_device_compiler_options += " -long-vector ";
     m_device_compiler_options += " -module-constants-in-storage-buffer ";
     m_device_compiler_options += " -cl-arm-non-uniform-work-group-size ";
-    m_device_compiler_options += " -enable-opaque-pointers=0 ";
 }
 
 void cvk_device::build_extension_ils_list() {
diff --git a/clvk/src/program.cpp b/clvk/src/program.cpp
index d52718d..66c0b06 100644
--- a/clvk/src/program.cpp
+++ b/clvk/src/program.cpp
@@ -932,8 +932,6 @@ cl_build_status cvk_program::do_build_inner_offline(bool build_to_ir,
         // Compose llvm-spirv command-line
         std::string cmd_spv{config.llvmspirv_bin};
         cmd_spv += " -r ";
-        cmd_spv +=
-            " -opaque-pointers=0 "; // FIXME(#380) Re-enable when clspv is ready
         cmd_spv += " -o ";
         cmd_spv += clspv_input_file;
         cmd_spv += " ";
@@ -1053,10 +1051,8 @@ cl_build_status cvk_program::do_build_inner_online(bool build_to_ir,
             std::lock_guard<std::mutex> llvmspirv_compile_lock(
                 llvmspirv_compile_mutex);
             int llvmArgc = 2;
-            const char* llvmArgv[2];
+            const char* llvmArgv[1];
             llvmArgv[0] = "llvm-spirv(online)";
-            llvmArgv[1] = "-opaque-pointers=0"; // FIXME(#380) Re-enable when
-                                                // clspv is ready
             llvm::cl::ResetAllOptionOccurrences();
             llvm::cl::ParseCommandLineOptions(llvmArgc, llvmArgv);
             if (!llvm::readSpirv(llvm_context, m_il_stream, llvm_module, err)) {
