diff --git a/clvk/src/program.cpp b/clvk/src/program.cpp
index 49029ed..e4596ed 100644
--- a/clvk/src/program.cpp
+++ b/clvk/src/program.cpp
@@ -1115,7 +1115,7 @@ cl_build_status cvk_program::do_build_inner(const cvk_device* device) {
     std::string tmp_folder;
     if (use_tmp_folder) {
         // Create temporary folder
-        std::string tmp_template{"clvk-XXXXXX"};
+        std::string tmp_template{"/tmp/clvk-XXXXXX"};
         const char* tmp = cvk_mkdtemp(tmp_template);
         if (tmp == nullptr) {
             return CL_BUILD_ERROR;
