diff --git a/clvk/src/device.cpp b/clvk/src/device.cpp
index d60af58..5223711 100644
--- a/clvk/src/device.cpp
+++ b/clvk/src/device.cpp
@@ -815,6 +815,7 @@ bool cvk_device::supports_capability(spv::Capability capability) const {
     case spv::CapabilityImage1D:
     case spv::CapabilityImageQuery:
     case spv::CapabilityImageBuffer:
+    case spv::CapabilitySampledBuffer:
         return true;
     // Optional capabilities:
     case spv::CapabilityFloat16:
