From 05f6e18fff7767f629098d11766661fad3d55f5c Mon Sep 17 00:00:00 2001
From: Kalvin Lee <kdlee@google.com>
Date: Mon, 21 Oct 2019 16:22:32 -0700
Subject: [PATCH] second commit

---
 cups-filters-1.21.6/filter/pdftoraster.cxx | 27 ----------------------
 1 file changed, 27 deletions(-)

diff --git a/filter/pdftoraster.cxx b/filter/pdftoraster.cxx
index 13229e1..f040b05 100644
--- a/filter/pdftoraster.cxx
+++ b/filter/pdftoraster.cxx
@@ -2174,30 +2174,3 @@ err1:
 
   return exitCode;
 }
-
-/* replace memory allocation methods for memory check */
-/* For compatibility with g++ >= 4.7 compilers _GLIBCXX_THROW
- *  should be used as a guard, otherwise use traditional definition */
-#ifndef _GLIBCXX_THROW
-#define _GLIBCXX_THROW throw
-#endif
-
-void * operator new(size_t size) _GLIBCXX_THROW (std::bad_alloc)
-{
-  return gmalloc(size);
-}
-
-void operator delete(void *p) throw ()
-{
-  gfree(p);
-}
-
-void * operator new[](size_t size) _GLIBCXX_THROW (std::bad_alloc)
-{
-  return gmalloc(size);
-}
-
-void operator delete[](void *p) throw ()
-{
-  gfree(p);
-}
-- 
2.23.0.866.gb869b98d4c-goog

