From 140394579dacfa4334d2aa2deb30cffc5fee609f Mon Sep 17 00:00:00 2001
From: Lepton Wu <ytht.net@gmail.com>
Date: Mon, 10 Feb 2020 15:27:31 -0800
Subject: [PATCH] Return error when no va devices available.

Signed-off-by: Lepton Wu <ytht.net@gmail.com>
---
 va/drm/va_drm_utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/va/drm/va_drm_utils.c b/va/drm/va_drm_utils.c
index 905ccdd..ac96fa2 100644
--- a/va/drm/va_drm_utils.c
+++ b/va/drm/va_drm_utils.c
@@ -69,7 +69,7 @@ VA_DRM_GetNumCandidates(VADriverContextP ctx, int * num_candidates)
     }
     drmFreeVersion(drm_version);
     *num_candidates = num_of_candidate;
-    return VA_STATUS_SUCCESS;
+    return num_of_candidate ? VA_STATUS_SUCCESS : VA_STATUS_ERROR_UNKNOWN;
 }
 
 /* Returns the VA driver name for the active display */
-- 
2.25.0.225.g125e21ebc7-goog

