From e3b073265b60bdcca23b8a0b16ef3687565cbde6 Mon Sep 17 00:00:00 2001
From: Zubin Mithra <zsm@chromium.org>
Date: Mon, 26 Apr 2021 09:41:45 -0700
Subject: [PATCH] cros-syzkaller: do not use go.sum and go.mod

---
 Makefile |   2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7c8d673ee..e1d5dea2b 100644
--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@ CGO_ENABLED ?= 0
 export CGO_ENABLED
 TARGETGOOS := $(TARGETOS)
 TARGETGOARCH := $(TARGETVMARCH)
-export GO111MODULE=on
+export GO111MODULE=off
 export GOBIN=$(shell pwd -P)/bin
 
 GITREV=$(shell git rev-parse HEAD)
