From 6acaae847fdddeb9e0f5bfbeb5ca4cb55aa216a1 Mon Sep 17 00:00:00 2001
From: Zubin Mithra <zsm@chromium.org>
Date: Mon, 18 Oct 2021 17:16:33 -0700
Subject: [PATCH 4/4] cros-syzkaller: only do exact function name matching

---
 pkg/mgrconfig/load.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/mgrconfig/load.go b/pkg/mgrconfig/load.go
index e021b69fe..934b4fffe 100644
--- a/pkg/mgrconfig/load.go
+++ b/pkg/mgrconfig/load.go
@@ -321,7 +321,7 @@ func ParseEnabledSyscalls(target *prog.Target, enabled, disabled []string) ([]in
 }
 
 func MatchSyscall(name, pattern string) bool {
-	if pattern == name || strings.HasPrefix(name, pattern+"$") {
+	if pattern == name {
 		return true
 	}
 	if len(pattern) > 1 && pattern[len(pattern)-1] == '*' &&
-- 
2.33.0.1079.g6e70778dc9-goog

