diff -uraN a/testsuite/backend/genesys/tests_image.cpp b/testsuite/backend/genesys/tests_image.cpp
--- a/testsuite/backend/genesys/tests_image.cpp	2020-02-02 04:58:09.000000000 -0700
+++ b/testsuite/backend/genesys/tests_image.cpp	2020-02-11 14:31:37.474894579 -0700
@@ -23,8 +23,8 @@
 #define DEBUG_DECLARE_ONLY
 
 #include "tests.h"
-#include "minigtest.h"
 #include "tests_printers.h"
+#include "minigtest.h"
 
 #include "../../../backend/genesys/image.h"
 #include "../../../backend/genesys/image_pipeline.h"
diff -uraN a/testsuite/backend/genesys/tests_image_pipeline.cpp b/testsuite/backend/genesys/tests_image_pipeline.cpp
--- a/testsuite/backend/genesys/tests_image_pipeline.cpp	2020-02-02 04:58:09.000000000 -0700
+++ b/testsuite/backend/genesys/tests_image_pipeline.cpp	2020-02-11 14:32:12.483027884 -0700
@@ -23,8 +23,8 @@
 #define DEBUG_DECLARE_ONLY
 
 #include "tests.h"
-#include "minigtest.h"
 #include "tests_printers.h"
+#include "minigtest.h"
 
 #include "../../../backend/genesys/image_pipeline.h"
 
diff -uraN a/testsuite/backend/genesys/tests_motor.cpp b/testsuite/backend/genesys/tests_motor.cpp
--- a/testsuite/backend/genesys/tests_motor.cpp	2020-02-02 04:58:09.000000000 -0700
+++ b/testsuite/backend/genesys/tests_motor.cpp	2020-02-11 14:32:43.715146964 -0700
@@ -23,8 +23,8 @@
 #define DEBUG_DECLARE_ONLY
 
 #include "tests.h"
-#include "minigtest.h"
 #include "tests_printers.h"
+#include "minigtest.h"
 
 #include "../../../backend/genesys/low.h"
 #include "../../../backend/genesys/enums.h"
diff -uraN a/testsuite/backend/genesys/tests_utilities.cpp b/testsuite/backend/genesys/tests_utilities.cpp
--- a/testsuite/backend/genesys/tests_utilities.cpp	2020-02-02 04:58:09.000000000 -0700
+++ b/testsuite/backend/genesys/tests_utilities.cpp	2020-02-11 14:32:58.915204971 -0700
@@ -23,8 +23,8 @@
 #define DEBUG_DECLARE_ONLY
 
 #include "tests.h"
-#include "minigtest.h"
 #include "tests_printers.h"
+#include "minigtest.h"
 
 #include "../../../backend/genesys/utilities.h"
 
