diff --git a/backend/epson2_usb.c b/backend/epson2_usb.c
index 87830c6..d956efa 100644
--- a/backend/epson2_usb.c
+++ b/backend/epson2_usb.c
@@ -176,6 +176,13 @@ SANE_Word sanei_epson_usb_product_ids[] = {
   0x8ca, /* L850 Series */
   0x8cd, /* WF-R4640 Series, WF-R5690 Series */
   0x8d0, /* PX-M350F, WF-M5690 Series */
+  0x1141, /*L3100 Series*/
+  0x1142, /*L3110 Series*/
+  0x110f, /*PX-M160T Series*/
+  0x1122, /*ET-2600 Series, ET-2610 Series, L3050 Series, L3060 Series,L395 Series,L396 Series */
+  0x1105, /*ET-2500 Series, L375 Series*/
+  0x1102, /*PX-048A Series, XP-230 Series,XP-235 Series*/
+  0x8d2,  /*L365 Series, L366 Series*/
   0				/* last entry - this is used for devices that are specified
 				   in the config file as "usb <vendor> <product>" */
 };
diff --git a/doc/descriptions/epson2.desc b/doc/descriptions/epson2.desc
index d1f3bed..ba1fe6b 100644
--- a/doc/descriptions/epson2.desc
+++ b/doc/descriptions/epson2.desc
@@ -3528,3 +3528,95 @@
 :interface  "USB"
 :usbid      "0x04b8" "0x08a7"
 :status     :good
+
+
+:model      "L3100 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1141"
+:status     :complete
+
+:model      "L3110 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1142"
+:status     :complete
+
+:model      "PX-M160T Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x110f"
+:status     :complete
+
+:model      "ET-2600 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "ET-2610 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "L3050 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "L3060 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "L395 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "L396 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1122"
+:status     :complete
+
+:model      "ET-2500 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1105"
+:status     :complete
+
+:model      "L375 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1105"
+:status     :complete
+
+:model      "PX-048A Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1102"
+:status     :complete
+
+:model      "XP-230 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1102"
+:status     :complete
+
+:model      "XP-235 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1102"
+:status     :complete
+
+:model      "PX-049A Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1116"
+:status     :complete
+
+:model      "XP-240 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x1116"
+:status     :complete
+
+:model      "L365 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x08d2"
+:status     :complete
+
+:model      "L366 Series"
+:interface  "USB"
+:usbid      "0x04b8" "0x08d2"
+:status     :complete
+
