diff --git a/clvk/tests/CMakeLists.txt b/clvk/tests/CMakeLists.txt
index afaa5b2..0b2d4b6 100644
--- a/clvk/tests/CMakeLists.txt
+++ b/clvk/tests/CMakeLists.txt
@@ -34,7 +34,7 @@ macro(add_gtest_executable name)
   include_directories(${CLVK_PROJECT_SOURCE_DIR}/src)
   include_directories(${Vulkan_INCLUDE_DIRS})
 
-  target_link_libraries(${name} OpenCL llvm_gtest llvm_gtest_main)
+  target_link_libraries(${name} OpenCL gtest gtest_main)
   if (NOT PROJECT_IS_TOP_LEVEL)
     if (${CLVK_VULKAN_IMPLEMENTATION} STREQUAL loader OR
         ${CLVK_VULKAN_IMPLEMENTATION} STREQUAL swiftshader)
