diff --git a/clvk/tests/CMakeLists.txt b/clvk/tests/CMakeLists.txt
index 76ceaf9..6494e69 100644
--- a/clvk/tests/CMakeLists.txt
+++ b/clvk/tests/CMakeLists.txt
@@ -12,13 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+cmake_minimum_required(VERSION 3.9)
+project(clvk_test)
 
-if (CLVK_COMPILER_AVAILABLE)
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/api)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/sha1)
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/simple)
-endif()
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/simple-from-binary)
 
 option(CLVK_BUILD_CONFORMANCE_TESTS "Build OpenCL conformance tests")
 if (CLVK_BUILD_CONFORMANCE_TESTS)
diff --git a/clvk/tests/api/CMakeLists.txt b/clvk/tests/api/CMakeLists.txt
index ae4e792..63189de 100644
--- a/clvk/tests/api/CMakeLists.txt
+++ b/clvk/tests/api/CMakeLists.txt
@@ -46,7 +46,7 @@ include_directories(${LLVM_BINARY_DIR}/include)
 include_directories(${LLVM_SOURCE_DIR}/include)
 include_directories(${LLVM_SOURCE_DIR}/utils/unittest/googletest/include/)
 
-target_link_libraries(${BINARY_NAME} OpenCL gtest gtest_main)
+target_link_libraries(${BINARY_NAME} OpenCL gtest gtest_main pthread)
 
 set_target_properties(${BINARY_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
     ${CMAKE_BINARY_DIR})
diff --git a/clvk/tests/simple/CMakeLists.txt b/clvk/tests/simple/CMakeLists.txt
index d921041..a628f68 100644
--- a/clvk/tests/simple/CMakeLists.txt
+++ b/clvk/tests/simple/CMakeLists.txt
@@ -13,16 +13,11 @@
 # limitations under the License.
 
 set(BINARY_NAME simple_test)
-set(BINARY_NAME_STATIC simple_test_static)
 
 add_executable(${BINARY_NAME} simple.cpp)
-add_executable(${BINARY_NAME_STATIC} simple.cpp)
 
 target_link_libraries(${BINARY_NAME} OpenCL)
-target_link_libraries(${BINARY_NAME_STATIC} OpenCL-static)
 
 set_target_properties(${BINARY_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
     ${CMAKE_BINARY_DIR})
-set_target_properties(${BINARY_NAME_STATIC} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
-    ${CMAKE_BINARY_DIR})
 
