From: Paul Moy <pmoy@chromium.org>
Zero-initialize m_pPrinterBuffer before use.

--- a/prnt/hpcups/HPCupsFilter.cpp
+++ b/prnt/hpcups/HPCupsFilter.cpp
@@ -510,7 +510,7 @@
         dbglog("HPCUPS: returning NO_ERROR from startPage\n");
     }
 
-    m_pPrinterBuffer = new BYTE[cups_header->cupsWidth * 4 + 32];
+    m_pPrinterBuffer = new BYTE[cups_header->cupsWidth * 4 + 32]();
 
     return NO_ERROR;
 }