diff --git a/Programs/defaults.h b/Programs/defaults.h
index 1c26219d8..7996f012b 100644
--- a/Programs/defaults.h
+++ b/Programs/defaults.h
@@ -80,7 +80,7 @@ extern "C" {
 
 #define DEFAULT_BRAILLE_KEYBOARD_ENABLED 1
 #define DEFAULT_BRAILLE_INPUT_MODE BRL_INPUT_TEXT
-#define DEFAULT_BRAILLE_QUICK_SPACE 0
+#define DEFAULT_BRAILLE_QUICK_SPACE 1
 
 #define DEFAULT_AUTORELEASE_TIME at20s
 #define DEFAULT_FIRST_RELEASE 1
