From ce55270cbb9fae7df8791a4dc51c960f63de0323 Mon Sep 17 00:00:00 2001
From: Po-Hsien Wang <pwang@chromium.org>
Date: Fri, 19 Apr 2019 11:24:09 -0700
Subject: [PATCH] doc install

---
 CMakeLists.txt | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 83c12d92..f0268e3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -602,20 +602,6 @@ endif ()
 ##############################################################################
 # Packaging
 
-install (
-    FILES
-        README.markdown
-        docs/BUGS.markdown
-        docs/NEWS.markdown
-        docs/USAGE.markdown
-    DESTINATION ${DOC_INSTALL_DIR}
-)
-install (
-    FILES LICENSE
-    DESTINATION ${DOC_INSTALL_DIR}
-    RENAME LICENSE.txt
-)
-
 set (CPACK_PACKAGE_VERSION_MAJOR "8")
 set (CPACK_PACKAGE_VERSION_MINOR "0")
 
-- 
2.21.0.392.gf8f6787159e-goog

