diff -Naur --show-c-function a/src/config.c b/src/config.c
--- a/src/config.c	2020-09-21 16:48:54.802774568 -0700
+++ b/src/config.c	2020-09-21 16:48:39.326821669 -0700
@@ -194,6 +194,16 @@ int loadConfig(char *configFile) {
             // Read next token...
             token = nextConfigToken();
             continue;
+        }
+        else if(strcmp("robustness", token)==0) {
+            // Got a robustness token...
+            token = nextConfigToken();
+            my_log(LOG_DEBUG, 0, "Config: Got robustness token : %s", token);
+            commonConfig.robustnessValue = atoi( token );
+
+            // Read next token...
+            token = nextConfigToken();
+            continue;
         } else {
             // Unparsable token... Exit...
             closeConfigFile();
