diff --git a/base/compiler_specific.h b/base/compiler_specific.h
index 327e3fa..1656564 100644
--- a/base/compiler_specific.h
+++ b/base/compiler_specific.h
@@ -235,4 +235,11 @@
 #define HAS_FEATURE(FEATURE) 0
 #endif
 
+// Macro for telling -Wimplicit-fallthrough that a fallthrough is intentional.
+#if defined(__clang__)
+#define FALLTHROUGH [[clang::fallthrough]]
+#else
+#define FALLTHROUGH
+#endif
+
 #endif  // BASE_COMPILER_SPECIFIC_H_
