Add a fuzzer target to meson.build.

http://b/160181793

diff -ruN a/meson.build b/meson.build
--- a/meson.build	2020-07-10 13:40:14.015406390 -0600
+++ b/meson.build	2020-07-10 13:48:30.282053752 -0600
@@ -1,4 +1,4 @@
-project('sane-airscan', 'c')
+project('sane-airscan', 'c', 'cpp')
 
 sources = [
   'airscan-array.c',
@@ -65,6 +65,13 @@
   install: true
 )
 
+executable(
+  'airscan_xml_fuzzer',
+  sources + ['airscan_xml_fuzzer.cc'],
+  dependencies: shared_deps,
+  build_by_default: false
+)
+
 install_man('sane-airscan.5')
 install_man('airscan-discover.1')
 install_data(['airscan.conf', 'dll.conf'], install_dir : 'etc/sane.d')
