diff --git a/test_conformance/select/test_select.cpp b/test_conformance/select/test_select.cpp
index 35f154a..05f566a 100644
--- a/test_conformance/select/test_select.cpp
+++ b/test_conformance/select/test_select.cpp
@@ -63,7 +63,7 @@ static void printUsage( void );
 //-----------------------------------------
 
 // Define the buffer size that we want to block our test with
-#define BUFFER_SIZE (1024*1024)
+#define BUFFER_SIZE (64*1024)
 #define KPAGESIZE 4096
 
 
