From 1910af3fc815afe64a3534c66102fd4018c5e992 Mon Sep 17 00:00:00 2001
From: Ryan T Huang <ryan.t.huang@intel.com>
Date: Wed, 13 Jan 2021 00:11:17 -0800
Subject: [PATCH] Enable new caps for VAConfigAttribEncryption

---
 test/test_data.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/test/test_data.h b/test/test_data.h
index 42a8b64..4ffda61 100644
--- a/test/test_data.h
+++ b/test/test_data.h
@@ -229,6 +229,15 @@ static const BitMasks g_vaFEIFunctionTypes = {
     VA_FEI_FUNCTION_ENC, VA_FEI_FUNCTION_PAK, VA_FEI_FUNCTION_ENC_PAK,
 };
 
+#if VA_CHECK_VERSION(1, 11, 0)
+static const BitMasks g_vaEncryptionTypes = {
+    VA_ENCRYPTION_TYPE_FULLSAMPLE_CTR,
+    VA_ENCRYPTION_TYPE_FULLSAMPLE_CBC,
+    VA_ENCRYPTION_TYPE_SUBSAMPLE_CTR,
+    VA_ENCRYPTION_TYPE_SUBSAMPLE_CBC,
+};
+#endif
+
 static const std::map<VAConfigAttribType, const BitMasks&>
     g_vaConfigAttribBitMasks = {
         {VAConfigAttribRTFormat, g_vaRTFormats},
@@ -237,6 +246,9 @@ static const std::map<VAConfigAttribType, const BitMasks&>
         {VAConfigAttribEncPackedHeaders, g_vaEncPackedHeaders},
         {VAConfigAttribEncInterlaced, g_vaEncInterlaced},
         {VAConfigAttribFEIFunctionType, g_vaFEIFunctionTypes},
+#if VA_CHECK_VERSION(1, 11, 0)
+        {VAConfigAttribEncryption, g_vaEncryptionTypes},
+#endif
     };
 
 static const BitMasks g_vaSurfaceMemTypes = {
-- 
2.25.1

