From 76f27802ef57cc777e6d8f96a8aff8b7cdbbd829 Mon Sep 17 00:00:00 2001
From: Ed Baker <edward.baker@intel.com>
Date: Wed, 10 Mar 2021 10:55:57 -0700
Subject: [PATCH] Temporarily remove VA-API 1.11.0 check

---
 test/test_data.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/test/test_data.h b/test/test_data.h
index 4ffda61..3963a73 100644
--- a/test/test_data.h
+++ b/test/test_data.h
@@ -229,14 +229,12 @@ static const BitMasks g_vaFEIFunctionTypes = {
     VA_FEI_FUNCTION_ENC, VA_FEI_FUNCTION_PAK, VA_FEI_FUNCTION_ENC_PAK,
 };
 
-#if VA_CHECK_VERSION(1, 11, 0)
 static const BitMasks g_vaEncryptionTypes = {
     VA_ENCRYPTION_TYPE_FULLSAMPLE_CTR,
     VA_ENCRYPTION_TYPE_FULLSAMPLE_CBC,
     VA_ENCRYPTION_TYPE_SUBSAMPLE_CTR,
     VA_ENCRYPTION_TYPE_SUBSAMPLE_CBC,
 };
-#endif
 
 static const std::map<VAConfigAttribType, const BitMasks&>
     g_vaConfigAttribBitMasks = {
@@ -246,9 +244,7 @@ static const std::map<VAConfigAttribType, const BitMasks&>
         {VAConfigAttribEncPackedHeaders, g_vaEncPackedHeaders},
         {VAConfigAttribEncInterlaced, g_vaEncInterlaced},
         {VAConfigAttribFEIFunctionType, g_vaFEIFunctionTypes},
-#if VA_CHECK_VERSION(1, 11, 0)
         {VAConfigAttribEncryption, g_vaEncryptionTypes},
-#endif
     };
 
 static const BitMasks g_vaSurfaceMemTypes = {
-- 
2.25.1

