From 570948c0a37374fcdfabb2e71c724ca62db12adc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 16 Jan 2019 17:55:49 +0200
Subject: [PATCH] client: Properly reset listener on error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 client/lxd_events.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/lxd_events.go b/client/lxd_events.go
index 15fc35cb..4cc660cd 100644
--- a/client/lxd_events.go
+++ b/client/lxd_events.go
@@ -81,7 +81,7 @@ func (r *ProtocolLXD) GetEvents() (*EventListener, error) {
 				}
 
 				// And remove them all from the list
-				r.eventListeners = []*EventListener{}
+				r.eventListeners = nil
 
 				conn.Close()
 				close(stopCh)
-- 
2.20.1.791.gb4d0f1c61a-goog

