commit 5d1fac6742418d9e69586ee27b09c851c2d59dd9
Author: Zhizhou Yang <zhizhouy@google.com>
Date:   Mon Feb 25 17:05:47 2019 -0800

    Link autofdo with libLLVM

diff --git a/m4/ax_llvm.m4 b/m4/ax_llvm.m4
index 194a7a2..274a2df 100644
--- a/m4/ax_llvm.m4
+++ b/m4/ax_llvm.m4
@@ -60,17 +60,10 @@ AC_DEFUN([AX_LLVM],
     if test -f "$ac_llvm_config_path"; then
       LLVM_CXXFLAGS=$($ac_llvm_config_path --cppflags)
       shared_mode=$($ac_llvm_config_path --shared-mode)
-      rpath=""
-      if test "x$shared_mode" = "xstatic"; then
-        LLVM_LDFLAGS="$($ac_llvm_config_path --ldflags) \
-                      $($ac_llvm_config_path --libs $1) \
-                      -ldl -lpthread -ltinfo"
-      elif test "x$shared_mode" = "xshared"; then
-        rpath="$($ac_llvm_config_path --libdir)"
-        LLVM_LDFLAGS="-Wl,-rpath $rpath \
-                      $($ac_llvm_config_path --ldflags) \
-                      $($ac_llvm_config_path --libs $1)"
-      fi
+      rpath="$($ac_llvm_config_path --libdir)"
+      LLVM_LDFLAGS="-Wl,-rpath $rpath \
+                   $($ac_llvm_config_path --ldflags) \
+                   -lLLVM"
 
       AC_REQUIRE([AC_PROG_CXX])
       CXXFLAGS_SAVED="$CXXFLAGS"
