Flatbuffers 1.12.0 introduces a dependency on absl/string_view.h and needs
to be explicitly defined in the build file.

diff --git a/tensorflow/lite/kernels/BUILD b/tensorflow/lite/kernels/BUILD
index 95f554ad..d0369e3b 100644
--- a/tensorflow/lite/kernels/BUILD
+++ b/tensorflow/lite/kernels/BUILD
@@ -395,6 +395,7 @@ cc_library(
         "//tensorflow/lite/kernels/internal:reference_base",
         "//tensorflow/lite/kernels/internal:tensor",
         "//tensorflow/lite/kernels/internal:tensor_utils",
+        "@com_google_absl//absl/strings:strings",
         "@farmhash_archive//:farmhash",
         "@flatbuffers",
     ],
