Remove unneeded repositories that fail to get downloaded correctly.

diff --git a/WORKSPACE b/WORKSPACE
index 9db1d9b8..9113db22 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -29,13 +29,6 @@ load("//third_party/toolchains/preconfig/generate:archives.bzl",
 
 bazel_toolchains_archive()
 
-load(
-    "@bazel_toolchains//repositories:repositories.bzl",
-    bazel_toolchains_repositories = "repositories",
-)
-
-bazel_toolchains_repositories()
-
 # Use `swift_rules_dependencies` to fetch the toolchains. With the
 # `git_repository` rules above, the following call will skip redefining them.
 load("@build_bazel_rules_swift//swift:repositories.bzl", "swift_rules_dependencies")
