diff -urp a/ESIF/Packages/Installers/chrome/dptf.conf b/ESIF/Packages/Installers/chrome/dptf.conf
--- ESIF/Packages/Installers/chrome/dptf.conf
+++ ESIF/Packages/Installers/chrome/dptf.conf
@@ -13,6 +13,13 @@ respawn
 script
 	DPTF_OPTS=""
 	DPTF_FILE="$(cros_config /thermal dptf-dv)" || true
+	if [ -z "${DPTF_FILE}" ]; then
+	  if [ -f "/etc/dptf/dptf_override.sh" ]; then
+	    . /etc/dptf/dptf_override.sh
+		DPTF_FILE="$(dptf_get_override)"
+	  fi
+	fi
+
 	if [ -n "${DPTF_FILE}" ]; then
 		dptf="/etc/dptf/${DPTF_FILE}"
 		if [ ! -f "${dptf}" ]; then
