From e52b208c76625da434f7f123679dfdbc78d11110 Mon Sep 17 00:00:00 2001
From: Chris Morin <cmtm@google.com>
Date: Wed, 5 Dec 2018 16:59:57 -0800
Subject: [PATCH] set values in journald.conf

---
 src/journal/journald.conf | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/journal/journald.conf b/src/journal/journald.conf
index 2f1c661153..ae09170302 100644
--- a/src/journal/journald.conf
+++ b/src/journal/journald.conf
@@ -12,24 +12,24 @@
 # See journald.conf(5) for details.
 
 [Journal]
-#Storage=auto
-#Compress=yes
-#Seal=yes
-#SplitMode=uid
+Storage=persistent
+Compress=no
+Seal=no
+SplitMode=none
 #SyncIntervalSec=5m
 #RateLimitIntervalSec=30s
 #RateLimitBurst=10000
-#SystemMaxUse=
+SystemMaxUse=200M
 #SystemKeepFree=
-#SystemMaxFileSize=
+SystemMaxFileSize=40M
 #SystemMaxFiles=100
 #RuntimeMaxUse=
 #RuntimeKeepFree=
 #RuntimeMaxFileSize=
 #RuntimeMaxFiles=100
 #MaxRetentionSec=
-#MaxFileSec=1month
-#ForwardToSyslog=no
+MaxFileSec=0
+ForwardToSyslog=yes
 #ForwardToKMsg=no
 #ForwardToConsole=no
 #ForwardToWall=yes
-- 
2.21.0.1020.gf2820cf01a-goog

