From ca2e7142b50e5ae4a7139e3cb2d81701bf437c11 Mon Sep 17 00:00:00 2001
From: Chris Morin <cmtm@google.com>
Date: Mon, 3 Dec 2018 17:49:00 -0800
Subject: [PATCH] support journald running without systemd

---
 src/journal/journald-server.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/journal/journald-server.c b/src/journal/journald-server.c
index 2a960ebb3e..4850252897 100644
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -1988,9 +1988,17 @@ int server_init(Server *s) {
         if (!s->rate_limit)
                 return -ENOMEM;
 
+
         r = cg_get_root_path(&s->cgroup_root);
-        if (r < 0)
-                return r;
+        if (r < 0) {
+                log_notice("Running without systemd.");
+                // Set cgroup_root to an empty string. We won't be able to use
+                // it since we're running without systemd.
+                s->cgroup_root = strdup("");
+                if (!s->cgroup_root)
+                        return -ENOMEM;
+        }
+
 
         server_cache_hostname(s);
         server_cache_boot_id(s);
-- 
2.21.0.1020.gf2820cf01a-goog

