From 87359bd1e88620956aafae33b709e2ae90981706 Mon Sep 17 00:00:00 2001
From: Sean Kau <skau@chromium.org>
Date: Wed, 5 Dec 2018 13:09:43 -0800
Subject: [PATCH 2/2] Prevent null dereference in fseek

---
 base/sfxstdio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/sfxstdio.c b/base/sfxstdio.c
index 140796ef6..eb8e33d18 100644
--- a/base/sfxstdio.c
+++ b/base/sfxstdio.c
@@ -126,7 +126,7 @@ s_file_read_seek(register stream * s, gs_offset_t pos)
         s->srptr = s->cbuf + offset - 1;
         return 0;
     }
-    if (pos < 0 || pos > s->file_limit ||
+    if (s->file == 0 || pos < 0 || pos > s->file_limit ||
         gp_fseek_64(s->file, s->file_offset + pos, SEEK_SET) != 0
         )
         return ERRC;
-- 
2.20.0.rc1.387.gf8505762e3-goog

