From aa05501cce0ecd78ec6207979e19b56d92b5aade Mon Sep 17 00:00:00 2001
From: Gert Wollny <gert.wollny@collabora.com>
Date: Wed, 21 Aug 2019 10:57:26 +0200
Subject: [PATCH] vrend: Unbind sampler view after creation

Signed-off-by: Gert Wollny <gert.wollny@collabora.com>
Reviewed-by: Gurchetan Singh <gurchetansingh@chromium.org>
---
 src/vrend_renderer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vrend_renderer.c b/src/vrend_renderer.c
index 372b805..3725413 100644
--- a/src/vrend_renderer.c
+++ b/src/vrend_renderer.c
@@ -2096,6 +2096,7 @@ int vrend_create_sampler_view(struct vrend_context *ctx,
            glTexParameteri(view->texture->target, GL_TEXTURE_SRGB_DECODE_EXT,
                             view->srgb_decode);
         }
+        glBindTexture(view->texture->target, 0);
      }
    }
 
-- 
2.23.0.187.g17f5b7556c-goog

