Do not regenerate locale archives.
See crbug.com/899377
--- a/extra/locale/locale-gen
+++ b/extra/locale/locale-gen
@@ -339,33 +339,6 @@
 [[ ${QUIET} -eq 0 ]] && [[ -z ${JUST_LIST} ]] && \
 einfo "Generation complete"
 
-if ${LOCALE_ARCHIVE} ; then
-	ebegin "Adding locales to archive"
-	# The pattern ends with / on purpose: we don't care about files (like
-	# locale-archive) in the locale subdir, and we definitely don't want to
-	# delete them!
-	for LOC in "${LOCALEDIR}"/*/; do 
-		LOC=${LOC%/} # Strip trailing /, since localedef doesn't like it
-		x=$(
-			"${DESTDIR}"usr/bin/localedef \
-				--add-to-archive "${LOC}" \
-				--replace \
-				--prefix "${DESTDIR%${EPREFIX}/}/"
-			ret=$?
-			if [[ -n ${output} ]] ; then
-				echo "${x}"
-			elif [[ ${ret} -ne 0 ]] ; then
-				eerror "${disp}: ${x}"
-			fi
-			if [[ $ret -eq 0 ]]; then
-				rm -r "${LOC}"
-			fi
-			exit ${ret}
-		)
-	done
-	eend $ret
-fi
-
 # Remove locales that existed but were not requested
 if [[ -n ${UPDATE} ]] ; then
 	# Ignore these pseudo locales
