diff --git a/src/libunwind/build.rs b/src/libunwind/build.rs
index 11c4c2faf1..042415aa81 100644
--- a/src/libunwind/build.rs
+++ b/src/libunwind/build.rs
@@ -18,7 +18,7 @@ fn main() {
         if target.contains("musl") {
             // musl is handled in lib.rs
         } else if !target.contains("android") {
-            println!("cargo:rustc-link-lib=gcc_s");
+            println!("cargo:rustc-link-lib=unwind");
         }
     } else if target.contains("freebsd") {
         println!("cargo:rustc-link-lib=gcc_s");
