diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index e8ec575ea37..94afa2349f7 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -760,7 +760,7 @@ fn supported_sanitizers(
         }
         "x86_64-apple-darwin" => darwin_libs("osx", &["asan", "lsan", "tsan"]),
         "x86_64-fuchsia" => common_libs("fuchsia", "x86_64", &["asan"]),
-        "x86_64-unknown-linux-gnu" => {
+        "x86_64-unknown-linux-gnu" | "x86_64-cros-linux-gnu" | "x86_64-pc-linux-gnu" => {
             common_libs("linux", "x86_64", &["asan", "lsan", "msan", "tsan"])
         }
         _ => Vec::new(),
