From f4630c411ddbd3a50d7d5781e726ca1dd9238d76 Mon Sep 17 00:00:00 2001
From: johnmach <john.machado@intel.com>
Date: Mon, 8 Mar 2021 13:46:59 +0530
Subject: [PATCH 2/4] Fix Null and Aritmatic Conversion Warning (#27)

---
 Source/GmmLib/TranslationTable/GmmPageTableMgr.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/GmmLib/TranslationTable/GmmPageTableMgr.cpp b/Source/GmmLib/TranslationTable/GmmPageTableMgr.cpp
index b24409d..0754fb1 100644
--- a/Source/GmmLib/TranslationTable/GmmPageTableMgr.cpp
+++ b/Source/GmmLib/TranslationTable/GmmPageTableMgr.cpp
@@ -412,7 +412,7 @@ ERROR_CASE:
 /////////////////////////////////////////////////////////////////////////////////////
 GMM_GFX_ADDRESS GmmLib::GmmPageTableMgr::GetAuxL3TableAddr()
 {
-    return AuxTTObj ? AuxTTObj->GetL3Address() : NULL;
+    return AuxTTObj ? AuxTTObj->GetL3Address() : 0ULL;
 }
 
 /////////////////////////////////////////////////////////////////////////////////////
@@ -471,7 +471,7 @@ GMM_STATUS GmmLib::GmmPageTableMgr::InitContextAuxTableRegister(HANDLE CmdQHandl
 /////////////////////////////////////////////////////////////////////////////////////
 GMM_STATUS GmmLib::GmmPageTableMgr::UpdateAuxTable(const GMM_DDI_UPDATEAUXTABLE *UpdateReq)
 {
-    if(GetAuxL3TableAddr() == NULL)
+    if(GetAuxL3TableAddr() == 0ULL)
     {
         GMM_ASSERTDPF(0, "Invalid AuxTable update request, AuxTable is not initialized");
         return GMM_INVALIDPARAM;
-- 
2.25.1

