From 1f676ccc12de5e0dcda27d413c60609a55b9f3ff Mon Sep 17 00:00:00 2001
From: johnmach <john.machado@intel.com>
Date: Mon, 8 Mar 2021 13:55:12 +0530
Subject: [PATCH 4/4] Initialize PageTable NodesPerTable inorder (#30)

---
 Source/GmmLib/TranslationTable/GmmUmdTranslationTable.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/GmmLib/TranslationTable/GmmUmdTranslationTable.h b/Source/GmmLib/TranslationTable/GmmUmdTranslationTable.h
index c774778..74de21d 100644
--- a/Source/GmmLib/TranslationTable/GmmUmdTranslationTable.h
+++ b/Source/GmmLib/TranslationTable/GmmUmdTranslationTable.h
@@ -453,8 +453,8 @@ namespace GmmLib
         GmmClientContext    *pClientContext;
 
         PageTable(int Size, int NumL3e, TT_TYPE flag) :
-            NodesPerTable(Size / PAGE_SIZE),
-            TTType(flag)
+            TTType(flag),
+	    NodesPerTable(Size / PAGE_SIZE)
         {
             InitializeCriticalSection(&TTLock);
 
-- 
2.25.1

