diff --git a/CPISPPlugIn/Makefile b/CPISPPlugIn/Makefile
index 12528c7..8658b22 100755
--- a/CPISPPlugIn/Makefile
+++ b/CPISPPlugIn/Makefile
@@ -12,7 +12,7 @@ DIR =$(notdir $(SRCS))
 OBJECTS_TAR = $(patsubst %cpp,%o,$(DIR))
 # Compiler flags
 
-CXX = clang++
+CXX ?= clang++
 CXXFLAGS += -I../CpComm/Comm_Export/ -I ../LinuxCommon -I../Tool_Common -Wall
 CXXFLAGS += -fPIC
 INSTALL_DIR=../ReleaseTool
diff --git a/CPISPTool/Makefile b/CPISPTool/Makefile
index 5131f80..d4946c7 100755
--- a/CPISPTool/Makefile
+++ b/CPISPTool/Makefile
@@ -6,7 +6,7 @@ INSTALL_DIR=../ReleaseTool
 SRCS = ISPTool.cpp CpRsString.cpp LinuxCommExport.cpp CpCommExport.cpp  CpIspExport.cpp LinuxIspExport.cpp \
 LinuxRsFuncBase.cpp CpRsFuncBase.cpp UnixParseArgs.cpp CpParseArgsBase.cpp
 
-CXX=clang++
+CXX ?= clang++
 CXXFLAGS += -I../CpComm/Comm_Export/ -I ../LinuxCommon -I../Tool_Common -I ./../CPISPPlugIn/CPISPCommon 
 
 DIR =$(notdir $(SRCS))
diff --git a/Comm_PlugIn/CommDevI2C/Makefile b/Comm_PlugIn/CommDevI2C/Makefile
index 03d3353..c06103f 100755
--- a/Comm_PlugIn/CommDevI2C/Makefile
+++ b/Comm_PlugIn/CommDevI2C/Makefile
@@ -1,28 +1,28 @@
-TARGET=libDevI2C
-VPATH = CommDevI2C ./../CPDevComCommon
-SRCS = LinuxDevI2C.cpp LinuxDevI2CRw.cpp CpDeviceComBase.cpp CpDeviceCommExtern.cpp
-DIR =$(notdir $(SRCS))
-OBJECTS_TAR := $(patsubst %cpp,%o,$(DIR))
-
-# Compiler flags
-
-CXX = clang++
-CXXFLAGS +=  -I../../Tool_Common/RsCommon/ -I../CPDevComCommon -Wall
-CXXFLAGS += -fPIC
-CXXFLAGS += -D DEVI2C
-LDFLAGS = 
-INSTALL_DIR=../../ReleaseTool/Comm
-
-all : $(TARGET)
-%.o:%.cpp
-	$(CXX) $(CXXFLAGS) -c $<
-$(TARGET): $(OBJECTS_TAR)
-	$(CXX) -shared -fPIC -o $(TARGET).so $(OBJECTS_TAR) $(CXXFLAGS) $(LDFLAGS)
-	$(AR) -rcs $(TARGET).a $(OBJECTS_TAR)
-
-install:
-	install -m 755 $(TARGET).so $(INSTALL_DIR)
-	#install -m 755 $(TARGET).a $(INSTALL_DIR)
-
-clean:
-	rm -rf *.o $(TARGET).so $(TARGET).a
+TARGET=libDevI2C
+VPATH = CommDevI2C ./../CPDevComCommon
+SRCS = LinuxDevI2C.cpp LinuxDevI2CRw.cpp CpDeviceComBase.cpp CpDeviceCommExtern.cpp
+DIR =$(notdir $(SRCS))
+OBJECTS_TAR := $(patsubst %cpp,%o,$(DIR))
+
+# Compiler flags
+
+CXX ?= clang++
+CXXFLAGS +=  -I../../Tool_Common/RsCommon/ -I../CPDevComCommon -Wall
+CXXFLAGS += -fPIC
+CXXFLAGS += -D DEVI2C
+LDFLAGS = 
+INSTALL_DIR=../../ReleaseTool/Comm
+
+all : $(TARGET)
+%.o:%.cpp
+	$(CXX) $(CXXFLAGS) -c $<
+$(TARGET): $(OBJECTS_TAR)
+	$(CXX) -shared -fPIC -o $(TARGET).so $(OBJECTS_TAR) $(CXXFLAGS) $(LDFLAGS)
+	$(AR) -rcs $(TARGET).a $(OBJECTS_TAR)
+
+install:
+	install -m 755 $(TARGET).so $(INSTALL_DIR)
+	#install -m 755 $(TARGET).a $(INSTALL_DIR)
+
+clean:
+	rm -rf *.o $(TARGET).so $(TARGET).a
diff --git a/CpComm/Makefile b/CpComm/Makefile
index 50092e2..b081210 100755
--- a/CpComm/Makefile
+++ b/CpComm/Makefile
@@ -7,7 +7,7 @@ DIR =$(notdir $(SRCS))
 OBJECTS_TAR = $(patsubst %cpp,%o,$(DIR))
 # Compiler flags
 
-CXX = clang++
+CXX ?= clang++
 CXXFLAGS += -I../Comm_PlugIn/Comm_PlugIn_Export/CPComm_PlugIn_Define -I ../LinuxCommon -I../Tool_Common -Wall
 CXXFLAGS += -fPIC
 INSTALL_DIR=../ReleaseTool
